### ISSUE TITLE
BSOD - btrfs.sys 

### ISSUE THREAD

**Issue by korodarn** [maharmstone/btrfs#644]
The driver used to work fine, but I'm not sure what has changed.  I am not sure how to diagnose this well on my end.  It has not caused any problems in Linux so far, I use the driver on game drives and such, and used to be able to play windows games on linux and vice versa thanks to sharing them on btrfs partition, but now, even if I'm not playing a game, updating, or otherwise touching files, I am getting this BSOD mentioning btrfs.sys 

Unfortunately it pops up quickly and it occurs early enough after I reboot that windows is basically not useable and doesn't give me a huge amount of time to diagnose or play around.  I did go ahead and install the latest updated debug driver, in case there was something more I could do to help diagnose the problem. 

I have run btrfs scrub in Linux on various partitions, but nothing seems to be coming up as a problem, and I'm not having any problems in Linux. 

Since I'm a primary Linux user this hasn't been a huge pain, enough to get me to reformat, but I've tried the basics of windows image repair and reinstalling the driver and those haven't worked.

**Comment by AndreasKarg** [comment]
I have run into this very same issue. Exactly the same symptoms - about 1 minute after boot I get a BSOD mentioning btrfs.sys. I haven't been able to get any details for the BSOD, as it doesn't seem to get logged in Windows' event log.

This started happening right after last week's round of Windows updates on Windows 11.
I _think_ at the time I was still on WinBtrfs 1.8 or so, and one of the things I tried to fix the issue was upgrading to 1.9, which made no difference.

I had a total of 3 BTRFS filesystems at the time, all distinct fs'es, each with one device in it:
1x pure data storage, mounted under Linux with zstd. Not sure if I ever configured WinBtrfs to also use compression. No subvolumes, no deduplication.
1x pure data storage, also mounted under Linux with zstd but not sure about Winbtrfs compression config. Lots of subvolumes, occasionally deduplicated using BEES.
1x Linux root with everything that goes with it, /home, /swap etc, also using zstd and lots of subvolumes. Auto-mounted by WinBtrfs but not accessed manually by myself. Occasionaly deduplicated using BEES.

The above setup had been working fine for ~2 years now, with daily switching between Linux and WIndows. After the crashes started, I ran a scrub on the second and third fs in the list (I'd completely forgotten that No.1 even was btrfs formatted) and found no issues.

I "solved" the issue by disabling WinBtrfs for now. For unrelated reasons, I had to recently reshuffle my filesystems so that now I only have the Linux root filesystem, spread across the original device + what had previously held fs No.2, merged into one with all data and subvolumes intact. I might re-enable WinBtrfs to see if it still crashes and report back.

**Comment by korodarn** [comment]
Yeah, sounds like we both were working well until we suddenly weren't, so likely some common cause in the windows updates.

**Comment by IceyMint** [comment]
I am unable to get any extra information but I am having the exact same issue. Only just started earlier this week. The same setup has been working for a long time. Same no issues when checking with btrfsck.

**Comment by bullmoos211** [comment]
Oddly, I found a different cause of this problem. I had both winbtrfs installed as well as totalcommander with few packages, but none of them included btrfs. Uninstalling Winbtrfs did not help me, but uninstalling totalcommander did. Now I reinstalled it without any extenisons and so far it works.
